backend/kernelCI_app/views/bootsView.py

```diff
@@ -0,0 +1,127 @@
+        parsedEnvMisc = None
+        if isinstance(misc_environment, dict):
+            parsedEnvMisc = misc_environment
+        else:
+            parsedEnvMisc = json.loads(misc_environment)
+        platform = parsedEnvMisc.get("platform")
+        if platform:
+            return platform
+        print("unknown platform in misc_environment", misc_environment)
+        return "unknown"
+
+    # TODO misc is not stable and should be used as a POC only
+    def extract_error_message(self, misc: Union[str, dict]):
+        parsedEnv = None
+        if isinstance(misc, dict):
+            parsedEnv = misc
+        else:
+            parsedEnv = json.loads(misc)
+        error_message = parsedEnv.get("error_msg")
+        if error_message:
+            return error_message
+        print("unknown error_msg in misc", misc)
+        return "unknown"
+
+    def get(self, request, commit_hash: str | None):
+        print("commit_hash", commit_hash)
+
+        names_map = {
+            "c.id": "id",
+            "c.git_repository_url": "git_repository_url",
+            "c.git_commit_hash": "git_commit_hash",
+            "t.build_id": "build_id",
+            "t.start_time": "start_time",
+            "t.status": "status",
+            "t.path": "path",
+            "b.architecture": "architecture",
+            "b.config_name": "config_name",
+            "b.compiler": "compiler",
+            "t.environment_misc": "environment_misc",
+            "t.environment_comment": "environment_comment",
+            "t.misc": "misc",
+        }
+
+        query = Checkouts.objects.raw(
+            """
+                SELECT DISTINCT ON (t.build_id) c.id, c.git_repository_url,
+                c.git_commit_hash, t.build_id, t.start_time,
+                t.status as status, t.path, b.architecture, b.config_name,
+                b.compiler, t.environment_misc, t.environment_comment, t.misc FROM checkouts AS c
+                INNER JOIN builds AS b ON c.id = b.checkout_id
+                INNER JOIN tests AS t ON t.build_id = b.id
+                WHERE c.git_commit_hash = %s AND t.path LIKE 'boot%%'
+                ORDER BY
+                build_id,
+                CASE t.status
+                    WHEN 'FAIL' THEN 1
+                    WHEN 'ERROR' THEN 2
+                    WHEN 'MISS' THEN 3
+                    WHEN 'PASS' THEN 4
+                ELSE 4
+                END,
+                t.id;
+            """,
+            [commit_hash],
+            translations=names_map,
+        )
+
+        errorCounts = {}
+        configCounts = {}
+        bootHistory = []
+        errorCountPerArchitecture = {}
+        platforms = set()
+        compilersPerArchitecture = {}
+        errorMessageCounts = {}
+        for record in query:
+            errorCounts[record.status] = errorCounts.get(record.status, 0) + 1
+            configCounts[record.config_name] = (
+                configCounts.get(record.config_name, 0) + 1
+            )
```

Comment:
One possibility is:
```python
from collections import defaultdict

# Initialize the defaultdict with int, which defaults to 0
errorCountPerArchitecture = defaultdict(int)
errorCountPerArchitecture[record.architecture] += 1

```

Reply:
https://github.com/kernelci/dashboard/pull/61/commits/db5a19c28bcf9c8186de01740beea10901e84386